path-manager/app/services/PathStore.scala

```diff
@@ -96,7 +96,71 @@ object PathStore extends Logging {
     }
   }
 
-  def updateCanonical(newPath: String, id: Long) = {
+  def registerAlias(proposedAliasPathRecord: PathRecord) = {
+
+    val id = proposedAliasPathRecord.identifier
+
+    logger.debug(s"Registering path [${proposedAliasPathRecord.path }] for [$id]")
+
+    if (!PathValidator.isValid(proposedAliasPathRecord.path)) {
+      Left(s"invalid path [${proposedAliasPathRecord.path}]")
+    } else {
+      val existingPath = Option(Dynamo.pathsTable.getItem("path", proposedAliasPathRecord.path)).map(PathRecord(_))
+
+      existingPath match {
+        case Some(pr) if (pr.identifier != id) => {
+          logger.warn(s"Failed to register existing path [${proposedAliasPathRecord.path}], already claimed by id [${pr.identifier}], submitting id [$id]")
+          Left("path already in use")
+        }
+        case _ => {
+          putPathItemAndAwaitIndexUpdate(proposedAliasPathRecord)
+
+          logger.debug(s"Registered new alias path [${proposedAliasPathRecord.path}] for id [$id] successfully")
+          Right(List(proposedAliasPathRecord).groupBy(_.`type`))
+        }
+      }
+    }
+  }
+
+  def updateCanonicalWithAlias(newPath: String, id: Long): Either[String, Map[String, List[PathRecord]]] = {
```

Comment:
I don't think this method name could be improved, but after discussion with @AWare, the concept of updating a canonical with an alias involves some logic that bore explanation. Could we add a comment to explain this process?